packages/project-access/src/project/ui5-config.ts

```diff
@@ -31,14 +31,38 @@ export async function getWebappPath(projectRoot: string, memFs?: Editor): Promis
  * @param projectRoot - path to project root
  * @param fileName - name of yaml file to be read
  * @param [memFs] - optional mem-fs editor instance
+ * @param options - options
+ * @param [options.validateSchema] - optional flag to validate the schema of the yaml file
  * @returns {UI5Config} UI5 config file in yaml format
  * @throws {Error} if file is not found
  */
-export async function readUi5Yaml(projectRoot: string, fileName: string, memFs?: Editor): Promise<UI5Config> {
+export async function readUi5Yaml(
+    projectRoot: string,
+    fileName: string,
+    memFs?: Editor,
+    options?: { validateSchema: boolean }
+): Promise<UI5Config> {
     const ui5YamlPath = join(projectRoot, fileName);
     if (await fileExists(ui5YamlPath, memFs)) {
         const yamlString = await readFile(ui5YamlPath, memFs);
-        return await UI5Config.newInstance(yamlString);
+        return await UI5Config.newInstance(yamlString, { validateSchema: options?.validateSchema });
     }
     throw Error(`File '${fileName}' not found in project '${projectRoot}'`);
 }
+
+/**
+ * Scans the project directory for ui5 configuration yaml files.
+ *
+ * @param projectRoot - path to project root, where ui5 configuration y*ml files are located
+ * @param [memFs] - optional mem-fs editor instance
+ * @returns list of valid and invalid UI5 configuration yaml file names
+ * @throws {Error} if an error occurs while reading files from projectRoot
+ */
+export async function getAllUi5YamlFileNames(projectRoot: string, memFs?: Editor): Promise<string[]> {
+    try {
+        const yamlFilePaths = await findFilesByExtension('.yaml', projectRoot, [], memFs, true);
+        return yamlFilePaths.map((path) => basename(path));
+    } catch (error) {
+        throw new Error(`There was an error reading files from the directory, '${projectRoot}': ${error}`);
```

Comment:
```suggestion
        throw new Error(`There was an error reading files from the directory '${projectRoot}': ${error}`);
```

Reply:
Done in f41acea6cdc7bcf341fd9c72b7a4306e0012a3b5